currency.sol

```diff
@@ -256,134 +281,176 @@ contract cccur is owned {
     // Prevent changing the Type of a super Admin (2) account
     if (accountType[_targetAccount] != 2){
         limitDebit[_targetAccount] = _debitLimit;
-        limitCredit[_targetAccount] = _creditLimit;
-        
+        limitCredit[_targetAccount] = _creditLimit;   
     }
+
     accountType[_targetAccount] = _accountType;
+    use(_targetAccount);
     
     emit SetAccountParams(now, _targetAccount, _accountStatus, accountType[_targetAccount], limitDebit[_targetAccount],  limitCredit[_targetAccount]);
     
     // ensure the ETH level of the account
+    refill();
     topUp(_targetAccount);
-    topUp(msg.sender);
   }
+
+  function AllowReplaceBy(address target) public payable {
+     if (!actif) revert();                                                      // panic lock
+     if (newAddress[msg.sender]!=address(0)) revert();                          // Already replaced account cannot be replaced again
+     if (!accountStatus[msg.sender]) revert();                                  // locked account cannot be replaced
+     if (accountAlreadyUsed[target]==true) revert();                            // only new account can be a replacement target
+     if (ReplacementRequestNumber[msg.sender]>2) revert();                      // limit the number of replacement request possible
+
+     if (requestReplacementTo[target]!=address(0)) {                            // Cancel replacement request if exists
+       requestReplacementFrom[requestReplacementTo[target]]=address(0);
+       requestReplacementTo[target]=address(0);
+     }
+
+     if (requestReplacementFrom[msg.sender]!=address(0)) {                      // Cancel replacement request if exists
+       requestReplacementTo[requestReplacementFrom[msg.sender]]=address(0);
+       requestReplacementFrom[target]=address(0);
+     }
+
+     requestReplacementFrom[target]=msg.sender;                                // register the request   
+     requestReplacementTo[msg.sender]=target;
```

Comment:
Not sure I understand correctly. Shouldn't this be the other way around ? If not, then the check in `acceptReplaceAccount()` is wrong. The third solution is that I didn't understand something. Let's recapitulate:

We want to change A -> B (Account A will be replaced by B), in your current proposal, is that what is supposed to happen ?:

A triggers `AllowReplaceBy(B)`
B triggers `AcceptReplace()`

If this is how it is supposed to happen, then my remark on these lines holds.

Reply:
outdated